objects/src/accounts/account_id.rs

```diff
@@ -134,336 +129,471 @@ impl FromStr for AccountStorageMode {
-    /// (normal account, fungible token, non-fungible token), the storage type (on/off chain), and
-    /// for the normal accounts if the code is updatable or not. These metadata bits are also
-    /// checked by the PoW and add to the total work defined below.
-    #[cfg(not(any(feature = "testing", test)))]
-    pub const REGULAR_ACCOUNT_SEED_DIGEST_MIN_TRAILING_ZEROS: u32 = 23;
-    #[cfg(not(any(feature = "testing", test)))]
-    pub const FAUCET_SEED_DIGEST_MIN_TRAILING_ZEROS: u32 = 31;
-    #[cfg(any(feature = "testing", test))]
-    pub const REGULAR_ACCOUNT_SEED_DIGEST_MIN_TRAILING_ZEROS: u32 = 5;
-    #[cfg(any(feature = "testing", test))]
-    pub const FAUCET_SEED_DIGEST_MIN_TRAILING_ZEROS: u32 = 6;
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
 
     /// Specifies a minimum number of ones for a valid account ID.
     pub const MIN_ACCOUNT_ONES: u32 = 5;
 
+    /// The serialized size of an [`AccountId`] in bytes.
+    pub const SERIALIZED_SIZE: usize = 15;
+
+    /// The lower two bits of the second least significant nibble determine the account type.
+    pub(crate) const TYPE_SHIFT: u64 = 4;
+    pub(crate) const TYPE_MASK: u64 = 0b11 << Self::TYPE_SHIFT;
+
+    /// The least significant nibble determines the account version.
+    const VERSION_MASK: u64 = 0b1111;
+
+    const ANCHOR_EPOCH_SHIFT: u64 = 48;
+    const ANCHOR_EPOCH_MASK: u64 = 0xffff << Self::ANCHOR_EPOCH_SHIFT;
+
+    /// The higher two bits of the second least significant nibble determine the account storage
+    /// mode.
+    pub(crate) const STORAGE_MODE_SHIFT: u64 = 6;
+    pub(crate) const STORAGE_MODE_MASK: u64 = 0b11 << Self::STORAGE_MODE_SHIFT;
+
+    pub(crate) const IS_FAUCET_MASK: u64 = 0b10 << Self::TYPE_SHIFT;
+
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a new account ID derived from the specified seed, code commitment and storage
-    /// commitment.
+    /// Creates an [`AccountId`] by hashing the given `seed`, `code_commitment`,
+    /// `storage_commitment` and `anchor_block_hash` and using the resulting first and second
+    /// element of the hash as the first and second felt of the ID. The given `anchor_epoch`
+    /// overwrites part of the second felt.
     ///
-    /// The account ID is computed by hashing the seed, code commitment and storage commitment and
-    /// using 1 element of the resulting digest to form the ID. Specifically we take element 0.
-    /// We also require that the last element of the seed digest has at least `23` trailing
-    /// zeros if it is a regular account, or `31` trailing zeros if it is a faucet account.
+    /// Note that the `anchor_epoch` and `anchor_block_hash` must correspond to a valid block in the
+    /// chain for the ID to be deemed valid during creation.
```

Comment:
Because there are many constructors for `AccountId` I tried to list the constraints once in the type documentation (under `# Constraints`) and reference it from each constructor in the `# Errors` section. Regarding the version it states:

> Constructors will return an error if:
> - The first felt contains account ID metadata (storage mode, type or version) that does not any of the known values.

which I think is fine. When looking at `AccountIdVersion` it should be obvious what the defined types are.